### ISSUE TITLE
Potential bug in add_task() function when assigning task IDs

### ISSUE THREAD

**Issue by MathBul** [noemiegs/ToDoList_Guisnel_CDOF1#5]
Hello,

In the provided code, there is a potential error in the add_task() function when adding a new task. The problem lies in the way the task ID is assigned.

Currently, the code uses the current length of the tasks list to determine the ID of the new task. However, this approach can be problematic if tasks are deleted, as this can result in duplicate or missing task IDs.

To correct this, you can use another method to assign task IDs, for example, using a separate counter that increments each time a task is added.

Here is a fixed version of the add_task() function with an alternative approach to assigning task IDs:

def add_task(tasks, description):
    task_id = 1
    while task_id in tasks:
        task_id += 1
    tasks[task_id] = {'description': description, 'completed': False}
    print(f"Task added with ID: {task_id}")

This change ensures that each task ID is unique, even if tasks have been previously deleted. You can use this updated version of add_task() function in your code.

**Comment by noemiegs** [comment]
You're absolutely correct about the potential issue in the add_task() function. Using the length of the tasks list to determine the task ID can indeed lead to conflicts, especially after deleting tasks. This can result in duplicate or missing task IDs. Your proposed solution to use a separate counter is a good approach to ensure unique IDs.

The implementation you've provided, where you iterate to find the next available ID, is a practical solution. You can commit your changes in the repo.
